src/Zenaton/v1/Runtime/Contexts/TaskContext.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Zenaton\Runtime\Contexts;
+
+/**
+ * Represents the current runtime context of a class.
+ *
+ * The information provided by the context can be useful to alter the behaviour of the task.
+ * For example, you can use the attempt index to know if a task has been automatically retried or not and how many times,
+ * and decide to do something when you did not expect the task to be retried more than X times.
+ * You can also use the attempt number in the `onErrorRetryDelay` method of a task in order to implement complex
+ * retry strategies.
+ */
+final class TaskContext
+{
```

Comment:
I tend to dislike `final` keyword. It'll even prevent us to inject a modified version. It's difficult to anticipate legitimate use cases.

Reply:
That's the point. Until we figure a legitimate use case, I prefer to avoid any unexpected usage / extend of this class.